### PR TITLE
Perplexity api

### PR DESCRIPTION
**Pull request by systemshift**
I have added Perplexity to the default AI models. I see Anthropic models are becoming part of the default as well, so I thought I should add a model that is specific for web search.

This pull request is a work in progress; reviews and edit recommendations are welcome.

### CODE REVIEW

File: pkg/wconfig/defaultconfig/presets/ai.json [hunk 1]
```diff
@@ -16,5 +16,13 @@
         "ai:model": "gpt-4o-mini",
         "ai:maxtokens": 2048,
         "ai:timeoutms": 60000
+    },
+    "ai@perplexity": {
```
Comment (esimkowitz) [inline pkg/wconfig/defaultconfig/presets/ai.json:20]: What happens if a user doesn't have a Perplexity token? Will this still work?

File: frontend/app/view/waveai/waveai.tsx [hunk 2]
```diff
@@ -173,47 +173,61 @@ export class WaveAiModel implements ViewModel {
...
                         noAction: true,
                     });
-                } else {
+                    break;
+                case "perplexity":
                     viewTextChildren.push({
                         elemtype: "iconbutton",
                         icon: "globe",
-                        title: "Using Remote Model @ " + baseUrl + " (" + modelName + ")",
+                        title: `Using Perplexity API (${aiOpts.model})`,
```
Comment (esimkowitz) [inline frontend/app/view/waveai/waveai.tsx:198]:
```suggestion
                        title: `Using Remote Perplexity API (${aiOpts.model})`,
```

**Review by esimkowitz — COMMENTED** [review]
Thank you very much for this submission, this is awesome!! Leaving one suggestion and a question.

**Comment by sawka** [comment]
This looks great.  I just got a perplexity API token and tested it and it completely worked!  We're about to build a new v0.10.0 beta, and I'd like to have this in.  Will need to merge main in order to pull.

**Comment by systemshift** [comment]
glad to see the pull request accepted. I was expecting more work though. In my opinion there should be some kind of option of routing via waveterm servers similar to how gpt mini does.
There are also some unique features in perplexity that require more work, when it provide links to back up facts, the links only show up as a reference but the user can't click or find them in the chat prompt.

**Comment by esimkowitz** [comment]
Ah sorry for jumping the gun, it worked for us so we assumed it was ready! Our code changes quickly and I didn't want to risk too many conflicts. Feel free to submit your improvements when you're ready.

**Pull request merged**